File: app/src/main/java/no/nav/pam/annonsemottak/annonsemottak/amedia/AmediaResponseMapper.java
package no.nav.pam.annonsemottak.annonsemottak.amedia;

import com.fasterxml.jackson.databind.JsonNode;
import com.google.common.collect.Lists;
import no.nav.pam.annonsemottak.stilling.Stilling;

import java.util.List;
import java.util.stream.Collectors;

/**
 * Mapper respons fra Amedia til en liste med stillinger
 */
class AmediaResponseMapper {

    static List<Stilling> mapResponse(
        JsonNode amediaResponse) {
        JsonNode hits = amediaResponse.path("hits").path("hits");
        List<JsonNode> hitlist = Lists.newArrayList(hits.iterator());

        return hitlist.stream()
            .map(h -> new AmediaStillingMapper(h).getStilling())
            .collect(Collectors.toList());
    }

    static List<String> mapEksternIder(JsonNode amediaResponse) {
        JsonNode hits = amediaResponse.path("hits").path("hits");
        List<JsonNode> hitlist = Lists.newArrayList(hits.iterator());

        return hitlist.stream()
            .map(h -> text(h.path("_id")))
            .collect(Collectors.toList());
    }

    static String text(JsonNode node) {
        return node == null || node.asText().equals("null") ? "" : node.asText();
    }

}

File: src/main/java/no/nav/pam/annonsemottak/receivers/amedia/AmediaResponseMapper.kt
package no.nav.pam.annonsemottak.receivers.amedia

import com.fasterxml.jackson.databind.JsonNode
import no.nav.pam.annonsemottak.stilling.Stilling

/**
 * Mapper respons fra Amedia til en liste med stillinger
 */
internal object AmediaResponseMapper {

    @JvmStatic
    fun mapResponse(amediaResponse: JsonNode): List<Stilling> = hitlist(amediaResponse)
            .map(::AmediaStillingMapper)
            .filter { !it.isFromNav }
            .map { it.getStilling() }

    @JvmStatic
    fun mapEksternIder(amediaResponse: JsonNode): List<String> = hitlist(amediaResponse)
            .map { h -> text(h.path("_id")) }

    private fun hitlist(amediaResponse: JsonNode) = amediaResponse.path("hits").path("hits")

    @JvmStatic
    fun text(node: JsonNode?) = node?.asText()?.convertNullToBlank() ?: ""

    private fun String.convertNullToBlank() = if(this == "null") "" else this
}
